utils/interfaces/_library/appsec_iast.py

```diff
@@ -0,0 +1,126 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+""" AppSec IAST validations """
+import traceback
+import json
+from collections import namedtuple
+from collections import Counter
+from utils.interfaces._core import BaseValidation
+from utils.interfaces._library._utils import get_spans_related_to_rid, get_rid_from_user_agent
+from utils.tools import m
+
+
+class _BaseAppSecIastValidation(BaseValidation):
+    """Base class for all IAST validations"""
+
+    path_filters = ["/v0.4/traces"]
+
+    def __init__(self, request):
+        super().__init__(request=request)
+        self.spans = []  # list of (trace_id, span_id): span related to rid
+        self.appsec_iast_events = []  # list of all appsec iast events
+
+    def check(self, data):
+        if data["path"] == "/v0.4/traces":
+            content = data["request"]["content"]
+
+            for i, span in enumerate(get_spans_related_to_rid(content, self.rid)):
+                self.log_debug(f'Found span with rid={self.rid}: span_id={span["span_id"]}')
+                self.spans.append(f'{span["trace_id"]}#{span["span_id"]}')
+
+                # I would like to make validations even if there aren't iast events
+                # if "_dd.iast.json" in span.get("meta", {}):
```

Comment:
"There is no vuln" use case must be treated by antoher class because of the nature of datadog spans, so you can filter here using `if "_dd.iast.json" in span`

For "no IAST event"  use case, the code will be : 

```python

class _NoIastEvent():
    is_success_on_expiry = True

    def validate(self, vulnerabilities):
        self.set_failure(f"{m(self.message)} => Vulnerabilites has been found : {vulnerabilities}")
```